lib/ast/expression.ts

```diff
@@ -279,6 +280,65 @@ export class InvocationExpression extends Expression {
     }
 }
 
+export class HistoryQueryExpression extends Expression {
+    selector : DeviceSelector;
+    channel : string;
+
+    constructor(location : SourceRange|null,
+                selector : DeviceSelector,
```

Comment:
Given you cannot pass an arbitrary device selector, but only a device kind, I would make HistoryQueryExpression store a `kind` property and avoid the DeviceSelector object.

Reply:
Done